control-plane/api/mesh/v2beta1/mesh_gateway_types.go

```diff
@@ -38,10 +39,16 @@ type MeshGateway struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
-	Spec   pbmesh.MeshGateway `json:"spec,omitempty"`
+	Spec   MeshGatewaySpec `json:"spec,omitempty"`
 	Status `json:"status,omitempty"`
 }
 
+type MeshGatewaySpec struct {
+	// GatewayClassName is the name of the GatewayClass used by the MeshGateway
+	GatewayClassName string                        `json:"gatewayClassName"`
+	Listeners        []*pbmesh.MeshGatewayListener `json:"listeners"`
```

Comment:
The `json` tags wind up differing over in the [generated Consul proto](https://github.com/hashicorp/consul/pull/20159/files#diff-790a8f054f856c26912a0292e2c2a3778edaf3ffc19460037b58eb50fa088ee3R34-R36), which is what we'll map this stuff into.

I think it might be best to just straight up copy these lines with their tags. Thoughts?

Reply:
I'll do that now, would be the best option I think